sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/json/JsonSuite.scala

```diff
@@ -2408,4 +2409,53 @@ class JsonSuite extends QueryTest with SharedSQLContext with TestJsonData {
       spark.read.option("mode", "PERMISSIVE").option("encoding", "UTF-8").json(Seq(badJson).toDS()),
       Row(badJson))
   }
+
+  test("SPARK-24204 error handling for unsupported data types") {
```

Comment:
Thank you for pinging me, @maropu .
Since this is all about file-based data sources, can we have all these test cases in `FileBasedDataSourceSuite`?

Reply:
The suite doesn't assume that all file-based data source has the same capability. In this PR, the test codes are almost the same and the only difference are the mapping tables. For example,
- `json` -> Interval
- `orc` -> Interval, Null
- `parquet` -> Interval, Null

Reply:
ok, I'll brush up the tests.